### PR TITLE
Home page

### LINKED ISSUE #2: Add: Home Page
Is your Feature Request Related to a Problem/a Blocker? Please Describe.
N/A

------

Describe How the Solution Should Behave
The home page component should be in the pages directory and the project cards should be created in the components directory. You can use some fake data to render the components.

--------

Screenshots of the Design/if any
(Add it after you finalise the figma design)

-------

Additional Context
N/A

### PR DESCRIPTION
**Pull request by haticecs**
- **Home page** component is added. 
- In the Home page **Header, Transation, FeaturedProjects, ProjectCard and Slider** components is used.
- Slider works with interval.
- Responsive design is added for the smaller screens.
- You can find the video recording.
[https://vimeo.com/697725679/b5b9d6fd6e](https://vimeo.com/697725679/b5b9d6fd6e)

### CODE REVIEW

File: src/components/ProjectCard/ProjectCard.jsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import PropTypes from 'prop-types';
+import featuredProjectImg from '../../assets/images/featuredProjectImg.png';
```
Comment (Iseluin) [inline src/components/ProjectCard/ProjectCard.jsx:3]: Please have a blank line between the global and local imports.

File: src/components/Slider/Slider.jsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import { useState, useEffect, useRef, useCallback } from 'react';
+import { IoIosArrowBack, IoIosArrowForward } from 'react-icons/io';
+import PropTypes from 'prop-types';
+
+export default function Slider({ items }) {
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  const slideRef = useRef();
+
+  // fucntion calculate the right index and sets it
```
Comment (Iseluin) [inline src/components/Slider/Slider.jsx]: This is a useful comment and we might keep it. Please correct the typo in it though. "Function that calculates and sets the right index"

File: src/components/Slider/Slider.jsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    (direction) => {
+      if (direction === 'left') {
+        setCurrentIndex(currentIndex > 0 ? currentIndex - 1 : items.length - 1);
+      } else {
+        setCurrentIndex(currentIndex < items.length - 1 ? currentIndex + 1 : 0);
+      }
+      slideRef.current.classList.add('fade-anim');
+    },
+    [currentIndex, items.length],
+  );
```
Comment (Iseluin) [inline src/components/Slider/Slider.jsx:21]: Loved the logic here.

File: src/components/Transation/Transation.test.jsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import { render, cleanup } from '@testing-library/react';
+import Transation from './Transation';
```
Comment (Iseluin) [inline src/components/Transation/Transation.test.jsx]: Blank line between please.

File: src/components/index.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import Header from './Header/Header';
+import Transation from './Transation/Transation';
+import FeaturedProjects from './FeaturedProjects/FeaturedProjects';
+import ProjectCard from './ProjectCard/ProjectCard';
+import Slider from './Slider/Slider';
+
+export { Header, Transation, FeaturedProjects, ProjectCard, Slider };
```
Comment (Iseluin) [inline src/components/index.js]: Those are all components related to the homepage so I guess it would be better if they would live inside a folder aptly named as HomePageExports or something like that, what do you think ? Because if we keep it like this, every other dev needs to export their components from here also.

File: src/constants/index.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+export const HOME_PAGE_TITLE = 'Dreams Will Come True';
+
+export const HOME_INFORMATION_TEXT =
+  'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minimveniam consequat.';
+
+export const TRANSATION_MAIN_TEXT =
+  'With the Right Organization, You Can Give A Hand True Needer';
+
+export const TRANSATION_SUB_TEXT =
+  "I'm a paragraph. Click here to add your own text and edit me. It’s easy. Just click “Edit Text” or double click me to add your own content and make changes to the font. Feel free to drag and drop me anywhere you like on your page. I’m a great place for you to tell a story and let your users know a little more about you.";
```
Comment (Iseluin) [inline src/constants/index.js]: Same goes here, should we take them inside a folder related to homepage constants or does the team want to use this way too ?

File: src/index.js [hunk 7]
```diff
@@ -2,14 +2,14 @@ import React from 'react';
...
+import Home from './pages/Home';
 import store from './app/store';
 import * as serviceWorker from './serviceWorker';
 
 ReactDOM.render(
   <React.StrictMode>
     <Provider store={store}>
-      <App />
+      <Home />
     </Provider>
```
Comment (Iseluin) [inline src/index.js]: Why did you delete the App component and rendered your Homepage component here? Do you think that we need to delete the App, if yes why do you think that we shouldn't render our components in the App but in index.js directly ?

File: src/setupTests.js [hunk 8]
```diff
@@ -2,4 +2,5 @@
 // allows you to do things like:
 // expect(element).toHaveTextContent(/react/i)
 // learn more: https://github.com/testing-library/jest-dom
```
Comment (Iseluin) [inline src/setupTests.js]: We don't need those comments I think.

**Review by Iseluin — CHANGES_REQUESTED** [review]
Great work @haticecs, I love the level of detail in this PR. 

1- There is a warning on the console about the prop types, please fix that.
2- App.test.jsx fails, I think because you deleted the App render in the index.js.
3- Can we make the primary hero to span the whole viewport height and width, that way the site will look more professional. 
4- I also have some questions on the tests you wrote which I will share with you later.
5- Will we add the button we discussed on the primary hero to introduce the users to the main functionality of the app? Where are we at this point as a team ?

File: src/components/Slider/Slider.jsx (hunk 3 above)
Comment (haticecs) [reply src/components/Slider/Slider.jsx:21]: thanks, I did slider by following tutorial :D

File: src/components/index.js (hunk 5 above)
Comment (haticecs) [reply src/components/index.js]: I totally agree, I am gonna change it 👍🏼

File: src/constants/index.js (hunk 6 above)
Comment (haticecs) [reply src/constants/index.js]: I updated them 👍🏼  We didn't discuss about it but we can talk today 👍🏼

File: src/index.js (hunk 7 above)
Comment (haticecs) [reply src/index.js]: I forget to delete it :( gonna update

**Review by maqalaqil — APPROVED** [review]
Amazing work @haticecs 
one of the best PRs i have seen so far. 
really good job! 
LGTM

**Comment by OzerOzturk** [comment]
You did a great job @haticecs by paying attention to responsive as well. I was also interested in your commit messages, I'm trying to adapt them to my work.

**Review by Iseluin — APPROVED** [review]
